windows/CodePushPackage.cs

```diff
@@ -0,0 +1,364 @@
+﻿using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System;
+using System.IO;
+using System.IO.Compression;
+using System.Threading;
+using System.Threading.Tasks;
+using Windows.Storage;
+using Windows.Storage.Streams;
+using Windows.Web.Http;
+
+namespace CodePush.ReactNative
+{
+    class CodePushPackage
+    {
+        private readonly string CODE_PUSH_FOLDER_PREFIX = "CodePush";
+        private readonly string CURRENT_PACKAGE_KEY = "currentPackage";
+        private readonly string DIFF_MANIFEST_FILE_NAME = "hotcodepush.json";
+        private readonly string DOWNLOAD_FILE_NAME = "download.zip";
+        private readonly string DOWNLOAD_URL_KEY = "downloadUrl";
+        private readonly string PACKAGE_FILE_NAME = "app.json";
+        private readonly string PACKAGE_HASH_KEY = "packageHash";
+        private readonly string PREVIOUS_PACKAGE_KEY = "previousPackage";
+        private readonly string RELATIVE_BUNDLE_PATH_KEY = "bundlePath";
+        private readonly string STATUS_FILE = "codepush.json";
+        private readonly string UNZIPPED_FOLDER_NAME = "unzipped";
+
+        public CodePushPackage()
+        {
+        }
+
+        private async Task<StorageFile> GetDownloadFile()
+        {
+            StorageFolder codePushFolder = await GetCodePushFolder();
+            return await codePushFolder.CreateFileAsync(DOWNLOAD_FILE_NAME, CreationCollisionOption.OpenIfExists);
+        }
+
+        private async Task<StorageFolder> GetUnzippedFolder()
+        {
+            StorageFolder codePushFolder = await GetCodePushFolder();
+            return await codePushFolder.CreateFolderAsync(UNZIPPED_FOLDER_NAME, CreationCollisionOption.OpenIfExists);
+        }
+
+        private async Task<StorageFolder> GetCodePushFolder()
+        {
+            return await ApplicationData.Current.LocalFolder.CreateFolderAsync(CODE_PUSH_FOLDER_PREFIX, CreationCollisionOption.OpenIfExists);
+        }
+
+        private async Task<StorageFile> GetStatusFile()
+        {
+            StorageFolder codePushFolder = await GetCodePushFolder();
+            return await codePushFolder.CreateFileAsync(STATUS_FILE, CreationCollisionOption.OpenIfExists);
+        }
+
+        public async Task<JObject> GetCurrentPackageInfo()
+        {
+            StorageFile statusFile = await GetStatusFile();
+            try
+            {
+                return await CodePushUtils.GetJObjectFromFile(statusFile);
+            }
+            catch (Exception e)
+            {
+                throw new CodePushUnknownException("Error getting current package info", e);
+            }
+        }
+
+        public async Task UpdateCurrentPackageInfo(JObject packageInfo)
```

Comment:
Can we sort the methods in this file? Also, I believe that many of the methods in this class are private. Can we update them as such?